Source/CardStackController.m

```diff
@@ -522,60 +511,86 @@ - (void)updateCardTitleBarBackgroundColors {
+            CGFloat springVelocity = (verticalVelocity * card.scale) * ((CGFloat)i / (CGFloat)(self.currentCardIndex + 1));
+            springAnimation.velocity = [NSValue valueWithCGRect:CGRectMake(0, springVelocity, 0, 0)];
+        }
+        [card pop_addAnimation:springAnimation forKey:@"frame"];
+    }
+}
+
+- (void)updateCardLocationsWhileOpening {
+    if (self.currentCardIndex > 0) {
+        CGFloat startY = CardStackTopMargin;
+        CardView *currentCard = [self.cards objectAtIndex:self.currentCardIndex];
+        CGFloat endY = currentCard.frame.origin.y;
+        CGFloat currentY = startY;
+        CGFloat incrementY = (endY - startY) / self.currentCardIndex;
+        for (NSUInteger i = 0; i < self.currentCardIndex; i++) {
+            currentCard = [self.cards objectAtIndex:i];
+            CGRect frame = currentCard.frame;
+            frame.origin.y = currentY;
+            currentCard.frame = frame;
+            currentY = currentY + incrementY;
+        }
+    }
+}
+
+- (CGRect)frameForCardAtIndex:(NSUInteger)index {
+    CGRect frame;
+
+    if (index <= self.currentCardIndex) {
         if (self.isOpen) {
-            CGFloat previousTitleBarHeights = 0.0f;
-            for (NSUInteger i = 0; i < self.cards.count; i++) {
+            CGFloat previousTitleBarHeights = CardStackTopMargin;
+            for (NSUInteger i = 0; i < index; i++) {
                 CardView *card = [self.cards objectAtIndex:i];
 
-                CGRect frame = card.frame;
-                if (i <= self.currentCardIndex) {
-                    frame.origin.y = previousTitleBarHeights + CardStackTopMargin;
-                } else {
-                    frame.origin.y = self.view.bounds.size.height - card.titleBarHeight;
-                }
-                card.frame = frame;
-
                 // -1.0f is used to avoid area below the title bar to become slightly visible is some cases (due to rounding errors)
                 previousTitleBarHeights += (card.titleBarHeight * card.scale - 1.0f);
             }
-        } else {
-            for (NSUInteger i = 0; i < self.cards.count; i++) {
-                CardView *card = [self.cards objectAtIndex:i];
 
-                CGRect frame = card.frame;
-                if (i <= self.currentCardIndex) {
-                    frame.origin.y = 0;
-                } else {
-                    frame.origin.y = self.view.bounds.size.height - card.titleBarHeight;
-                }
-                card.frame = frame;
-            }
+            CardView *card = [self.cards objectAtIndex:index];
+            frame = card.frame;
+            frame.origin.y = previousTitleBarHeights;
+        } else {
+            CardView *card = [self.cards objectAtIndex:index];
+            frame = card.frame;
+            frame.origin.y = 0;
         }
+    } else if (index > self.currentCardIndex && index < self.cards.count - 1) {
```

Comment:
Would be better if you named this `if` instead of adding comments bellow

```objc
BOOL thisShouldBeDoneBecause = (....);
if (thisShouldBeDoneBecause) {
    // ...
} else {
    // ...
}
```